sdk/src/pubkey.rs

```diff
@@ -104,40 +104,38 @@ pub fn read_pubkey_file(infile: &str) -> Result<Pubkey, Box<dyn error::Error>> {
     Ok(Pubkey::from_str(&printable)?)
 }
 
+/// Convenience macro to declare a static Pubkey and functions to interact with it
+///
+/// bs58_string: bs58 string representation the program's id
+///
+/// # Examples
+///
+/// ```
+/// solana_sdk::declare_id!("My!!!11111111111111111111111111111111111111");
+/// ```
 #[macro_export]
-macro_rules! solana_id(
-    ($id:ident) => (
+macro_rules!
+declare_id(
+    ($bs58_string:expr) => (
+        use std::str::FromStr;
+
+        $crate::lazy_static::lazy_static! {
+            static ref _PUBKEY: $crate::pubkey::Pubkey = { $crate::pubkey::Pubkey::from_str(&$bs58_string).unwrap() };
+        }
 
         pub fn check_id(id: &$crate::pubkey::Pubkey) -> bool {
-            id.as_ref() == $id
+            *id == *_PUBKEY
```

Comment:
id == _PUBKEY faster?

Reply:
I benched, this solution is the same speed as the original